### ISSUE TITLE
Are interfaces possible?

### ISSUE THREAD

**Issue by kitbs** [mghoneimy/php-graphql-client#6]
Is it possible to use interfaces with this library? For example the `... on Tree` syntax in the following snippet from the GitHub API:

```graphql
query {
  repository(owner: "vendor", name: "package") {
    object(expression: "master:src/") {
      ... on Tree {
        entries {
          name
        }
      }
    }
  }
}
```

Thanks for a really useful package!

**Comment by mghoneimy** [comment]
@kitbs not yet, I'm planning to work on that right after adding support for variables. Should take 2-3 weeks to get this done given my current schedule.

**Comment by mghoneimy** [comment]
@kitbs Support for line fragments has been released today on v1.3
https://github.com/mghoneimy/php-graphql-client/releases/tag/v1.3

You can checkout examples on how to use it here as well:
https://github.com/mghoneimy/php-graphql-client#using-interfaces-query-with-inline-fragments
